emcc.py

```diff
@@ -1781,10 +1781,14 @@ def set_max_memory():
       diagnostics.warning('unused-command-line-argument', 'MAXIMUM_MEMORY is only meaningful with ALLOW_MEMORY_GROWTH')
     settings.MAXIMUM_MEMORY = settings.INITIAL_MEMORY
 
-  # INITIAL_MEMORY sets a lower bound for MAXIMUM_MEMORY
   if 'MAXIMUM_MEMORY' not in user_settings:
-    if settings.INITIAL_MEMORY > settings.MAXIMUM_MEMORY:
-      settings.MAXIMUM_MEMORY = settings.INITIAL_MEMORY
+    if settings.ALLOW_MEMORY_GROWTH:
+      if settings.INITIAL_MEMORY > 2 * 1024 * 1024 * 1024:
+        settings.MAXIMUM_MEMORY = 4 * 1024 * 1024 * 1024
```

Comment:
It seems like this disallows an initial size of 2.01 GB that is growable up to 3 GB (but no more)?

Reply:
This entire block only runs if the user doesn't specify their own `MAXIMUM_MEMORY`.

Reply:
I see, thanks.